packages/babel-types/src/modifications/typescript/removeTypeDuplicates.js

```diff
@@ -0,0 +1,89 @@
+import {
+  isTSAnyKeyword,
+  isTSUnionType,
+  isTSBooleanKeyword,
+  isTSBigIntKeyword,
+  isTSNeverKeyword,
+  isTSNullKeyword,
+  isTSNumberKeyword,
+  isTSObjectKeyword,
+  isTSStringKeyword,
+  isTSUnknownKeyword,
+  isTSVoidKeyword,
+  isTSUndefinedKeyword,
+  isTSLiteralType,
+  isTSThisType,
+} from "../../validators/generated";
+
+/**
+ * Dedupe type annotations.
+ */
+export default function removeTypeDuplicates(
+  nodes: Array<Object>,
+): Array<Object> {
+  const generics = {};
+  const bases = {};
+
+  // store union type groups to circular references
+  const typeGroups = [];
+
+  const types = [];
+
+  for (let i = 0; i < nodes.length; i++) {
+    const node = nodes[i];
+    if (!node) continue;
+
+    // detect duplicates
+    if (types.indexOf(node) >= 0) {
+      continue;
+    }
+
+    // this type matches anything
+    if (isTSAnyKeyword(node.type)) {
+      return [node];
+    }
+
+    // Basic TS types, analogue of FlowBaseAnnotation
+    if (
```

Comment:
I'm not sure if I need to add custom type for `TSBaseKeywords`

Reply:
Yeah we can definitely add an alias similar to `FlowBaseAnnotation`.

Reply:
I added `TSBaseType`